refinery/static/source/js/refinery/solr/solr_response.js

```diff
@@ -60,94 +60,96 @@ SolrResponse.prototype.getDocumentList = function() {
 
 
 SolrResponse.prototype._processFacetCounts = function ( response ) {
-	
+
 	var self = this;
-	
+
 	var facetCounts = {};
-	
-	for ( var facet in response.facet_counts.facet_fields ) {
-		if ( response.facet_counts.facet_fields.hasOwnProperty( facet ) ) {
-			
-			if ( !facetCounts.hasOwnProperty[facet] ) {
-				facetCounts[facet] = {};
-			}
-							
-			for ( var j = 0; j < response.facet_counts.facet_fields[facet].length; j += 2 ) {
-				var facetValue = response.facet_counts.facet_fields[facet][j];
-				var facetValueCount = response.facet_counts.facet_fields[facet][j+1];
-				
-				if ( ( facetValue === "" ) || ( facetValue === undefined ) ) {
-					facetValue = "undefined";
+	if (response.facet_counts) {
+		for (var facet in response.facet_counts.facet_fields) {
+			if (response.facet_counts.facet_fields.hasOwnProperty(facet)) {
+
+				if (!facetCounts.hasOwnProperty[facet]) {
+					facetCounts[facet] = {};
+				}
+
+				for (var j = 0; j < response.facet_counts.facet_fields[facet].length; j += 2) {
+					var facetValue = response.facet_counts.facet_fields[facet][j];
+					var facetValueCount = response.facet_counts.facet_fields[facet][j + 1];
+
+					if (( facetValue === "" ) || ( facetValue === undefined )) {
+						facetValue = "undefined";
+					}
+
+					facetCounts[facet][facetValue] = facetValueCount;
 				}
-				
-				facetCounts[facet][facetValue] = facetValueCount;
 			}
-		}		
-    }
-    
+		}
+	}
+
     return facetCounts;
 }
 
 
 SolrResponse.prototype._processFieldList = function ( response ) {
 	var self = this;
-	
+
 	var fieldList = [];
 
 	for ( var i = 0; i < response.response.docs.length; ++i ) {
 		var document = response.response.docs[i];
-		
+
 		for ( var field in document ) {
 			if ( document.hasOwnProperty( field ) ) {
 				fieldList.push( field );
 			}
 		}
-		
+
 		// TODO: if all documents should be processed change this to get union of all fields
-		break;		
+		break;
 	}
-	
+
 	return fieldList;
 }
 
 
 SolrResponse.prototype._processDocumentList = function ( response ) {
 	var self = this;
-	
+
 	var documentList = [];
-	
+
 	for ( var i = 0; i < response.response.docs.length; ++i ) {
 		documentList.push( response.response.docs[i] );
     }
-    
+
     return documentList;
 }
 
 
 SolrResponse.prototype._processPivotCounts = function ( response ) {
```

Comment:
self._pivotCounts = self._processPivotCounts(response)
The default value for self._pivotCount = {}
I'd recommend setting an else with {}, even though no extra errors seem to be thrown.